tests/mashupDocumentParser.test.ts

```diff
@@ -0,0 +1,36 @@
+import MashupHandler from "../src/mashupDocumentParser";
+import { arrayUtils, pqUtils } from "../src/utils";
+import { simpleQuery, section1mMock } from "./mocks";
+import base64 from "base64-js";
+import JSZip from "jszip";
+import WorkbookTemplate from "../src/workbookTemplate";
+import { section1mPath } from "../src/constants";
+describe("Mashup Document Parser tests", () => {
+    test("ReplaceSingleQuery test", async () => {
+        const mashupHandler = new MashupHandler();
+
+        const defaultZipFile = await JSZip.loadAsync(
+            WorkbookTemplate.SIMPLE_QUERY_WORKBOOK_TEMPLATE,
+            { base64: true }
+        );
+        const base64str = await pqUtils.getBase64(defaultZipFile);
+
+        const replacedQueryStr = await mashupHandler.ReplaceSingleQuery(
+            base64str,
+            simpleQuery
+        );
+
+        const buffer = base64.toByteArray(replacedQueryStr).buffer;
+        const mashupArray = new arrayUtils.ArrayReader(buffer);
+        const startArray = mashupArray.getBytes(4);
+        const packageSize = mashupArray.getInt32();
+        const packageOPC = mashupArray.getBytes(packageSize);
+
+        const zip = await JSZip.loadAsync(packageOPC);
+        const section1m = await zip.file(section1mPath)?.async("text");
+
+        expect(section1m.replace(/ /g, "")).toEqual(
```

Comment:
why is the replacement required?

Reply:
the mock doesn't exactly match the output regarding whitespace cuz it has multiple lines. so this is to make sure the important part (Query) is matching.

Reply:
Can't we simpliy adjust the mock then?

Reply:
It was painful trying to adjust the whitespaces in the multi-line template string.